src/deepstream_pipeline_background_removal_cpu.py

```
@@ -51,43 +52,43 @@ def _map_bgr_from_buffer(pad: Gst.Pad, buf: Gst.Buffer) -> np.ndarray | None:
             stride = int(meta.stride[0])
         else:
-            caps = pad.get_current_caps().get_structure(0)
-            width = int(caps.get_value("width"))
-            height = int(caps.get_value("height"))
+            s = pad.get_current_caps().get_structure(0)
+            width = int(s.get_value("width"))
+            height = int(s.get_value("height"))
             stride = width * 3
+
         arr = np.frombuffer(map_info.data, dtype=np.uint8)
-        expected = height * stride
-        if arr.size >= expected:
-            frame2d = arr[:expected].reshape((height, stride))
-            if stride == width * 3:
-                frame = frame2d.reshape((height, width, 3))
-            else:
-                frame = frame2d[:, : width * 3].reshape((height, width, 3)).copy()
-        else:
-            frame = arr.reshape((height, width, 3))
+        frame = arr.reshape((height, stride))[:, : width * 3].reshape((height, width, 3))
         return frame
+    except Exception:
+        return None
     finally:
         buf.unmap(map_info)
 
 
-class _Ctx:
+class BackgroundRemovalCtx:
+    """Context passed to the probe."""
+
     def __init__(self, out_dir: str, save_every: int):
-        self.out_dir = out_dir
-        self.save_every = max(1, save_every)
-        self.counter = 0
+        self.out_dir: str = out_dir
+        self.save_every: int = max(1, save_every)
+        self.counter: int = 0
 
 
-def background_removal_probe(pad, info, u_data: _Ctx):
+def background_removal_probe(pad: Gst.Pad, info: Gst.PadProbeInfo, u_data: BackgroundRemovalCtx) -> Gst.PadProbeReturn:
     buf = info.get_buffer()
     if not buf:
-        return Gst.PadProbeReturn.OK
+        print("[probe][error] missing GstBuffer; dropping")
+        return Gst.PadProbeReturn.DROP  # avoid silent OK
 
     frame = _map_bgr_from_buffer(pad, buf)
     if frame is None:
-        return Gst.PadProbeReturn.OK
+        print("[probe][error] failed to map frame; dropping")
```

It's better to throw an error/exception